app/packages/looker/src/lookers/abstract.ts

```diff
@@ -515,7 +515,48 @@ export abstract class AbstractLooker<
   abstract updateOptions(options: Partial<State["options"]>): void;
 
   updateSample(sample: Sample) {
-    this.loadSample(sample);
+    // collect any mask targets array buffer that overlays might have
+    // we'll transfer that to the worker instead of copying it
+    const arrayBuffers: ArrayBuffer[] = [];
+
+    for (const overlay of this.pluckedOverlays ?? []) {
```

Comment:
I think `sampleOverlays` should be used here. `pluckedOverlays` contains the current frame overlays, and that is handled separately in VideoLooker.updateSample

Reply:
Yeah... although we'd have to _copy_ the bitmaps to the worker and I'd much rather avoid that. In my next PR, where I'll be working on async overlay loading, we'll be adding a "loading" state, maybe that'll make it better?